sources/core/objects/Interface.cpp

```diff
@@ -128,35 +138,60 @@ void HumanInterface::UpdateHealth()
     if (IdPtr<Human> human = owner_)
     {
         int health = human->GetHealth();
-        if (health == 100)
-        {
-            health_.GetView()->SetState("health0");
-        }
-        else if (health >= 80)
-        {
-            health_.GetView()->SetState("health1");
-        }
-        else if (health >= 60)
-        {
-            health_.GetView()->SetState("health2");
-        }
-        else if (health >= 40)
-        {
-            health_.GetView()->SetState("health3");
-        }
-        else if (health >= 20)
-        {
-            health_.GetView()->SetState("health4");
-        }
-        else if (health >= 0)
-        {
-            health_.GetView()->SetState("health5");
-        }
-        else //if (health_.GetView()->GetBaseFrameset()->GetState() != "health6")
+        const int STATES_AMOUNT = 7;
+        int state = (health * STATES_AMOUNT) / (HUMAN_MAX_HEALT + 1) > 0 ? (health * STATES_AMOUNT) / (HUMAN_MAX_HEALT + 1) : 0; // it will be from 0 to 6
```

Comment:
Again. ternary operator is not really suitable here.
```
 int state = qMax(0, (health * STATES_AMOUNT) / (HUMAN_MAX_HEALTH + 1))
```
will be much shorter and legible.